Atomic/util.py

```diff
@@ -842,21 +842,31 @@ def wrapper(*args, **kwargs):
 
 
 class InstallData(object):
-    if not os.path.exists(ATOMIC_INSTALL_JSON):
-        open(ATOMIC_INSTALL_JSON, 'a').close()
-
-    install_file_handle = open(ATOMIC_INSTALL_JSON, 'r')
 
     @staticmethod
-    def _read_install_data(file_handle):
+    def get_install_file_handle():
+        install_file_handle = open(ATOMIC_INSTALL_JSON, 'r')
+        return install_file_handle
+
+    @classmethod
+    def _read_install_data(cls):
         try:
-            return json.loads(file_handle.read())
+            file_handle = cls.get_install_file_handle()
+        except IOError:
+            return {}
+        try:
+            install_data = json.loads(file_handle.read())
+            file_handle.close()
+            return install_data
         except ValueError:
             return {}
 
     @classmethod
     def _write_install_data(cls, new_data):
-        install_data = cls._read_install_data(cls.install_file_handle)
+        try:
+            install_data = cls._read_install_data()
+        except IOError:
```

Comment:
This is already caught by `_read_install_data`, no? So you should already be getting `{}` back in that case.